### PR TITLE
Queue warning

### PR DESCRIPTION
**Pull request by hjoliver**
These changes partially address #3470 (close after PR port to 7.9.x and master as well)

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).

- [x] Appropriate tests are included (unit and/or functional).
- [x] Appropriate change log entry included.

- [x] No documentation update required. (No significant change to users, just fewer warnings)

### CODE REVIEW

**Comment by hjoliver** [comment]
```
[scheduling]
   [[queues]]
       [[[q1]]]
           members = A, B
       [[[q2]]]
           members = x
   [[dependencies]]
       graph = "x => y"
[runtime]
   [[A]]
   [[B]]
   [[x]]
       inherit = A, B
   [[y]]
```
On 7.8.x:
```
cylc-flow-7.8.x$ cylc val foo
WARNING - Queue configuration warnings:
        + q1: ignoring x from B (already assigned to a queue)  <--- SPURIOUS
        + q2: ignoring x (already assigned to a queue)  <--- CORRECT
Valid for cylc-7.8.4-21-gb766b-dirty
```
On this PR branch:
```
cylc-flow-7.8.x$ cylc val foo
WARNING - Queue configuration warnings:
        + q2: ignoring x (already assigned to a queue)  <--- CORRECT
Valid for cylc-7.8.4-22-g9db758
```

**Comment by kinow** [comment]
Conflicts in `CHANGES.md` @hjoliver

**Comment by hjoliver** [comment]
> Conflicts in `CHANGES.md` @hjoliver

(just rebased)

**Comment by hjoliver** [comment]
I'll replace the new functional test with a unit test, on the master version of this (once this is approved).

**Comment by hjoliver** [comment]
Note as a follow-on to this change I could change the existing behaviour (as marked `<---- CORRECT` in the example above) to arguably more standard override instead of warn and ignore. i.e. if a task gets assigned to two queues, the second assignment sticks, not the first. However, I'm unsure if anyone is relying on the current behaviour - what do you think @oliver-sanders ? 

Note this is discussed here along with the possibility of allowing tasks to belong to multiple queues at once, but I don't think we figured out a sensible way of doing that: #2701

File: lib/cylc/config.py [hunk 1]
```diff
@@ -1193,31 +1196,32 @@ def configure_queues(self):
...
                 queue['members'] = qmembers
             else:
                 del queues[key]
 
+        if warnings:
+            err_msg = "Queue configuration warnings:"
+            for msg in warnings:
+                err_msg += "\n+ %s" % msg
+            LOG.warning(err_msg)
+
```
Comment (hjoliver) [inline lib/cylc/config.py]: (This block was wrongly inside the outermost loop, resulting in repeated warnings!)

**Pull request merged**